generator/src/routes/+page.svelte

```diff
@@ -98,12 +103,29 @@
     document.body.removeChild(link)
   }
 
-  function onChangeValidSignature(event: Event) {
-    const target = event.target as HTMLInputElement
-    if (target.checked) {
+  function clearError() {
+    error = null
+  }
+  function onChangeValidSignature() {
+    if (isValid) {
       issValue = settings.ISS_DOMAIN
     }
   }
+  function onChangeProductType() {
+    productId = ""
+    product = ""
+  }
+  function onChangeProductId() {
+    if (productType === ProductType.PREMADE) {
+      product = premadeProducts.find((product) => product.id === productId)?.product!
+    }
+  }
+
+  $: isValid, onChangeValidSignature(), clearError()
+  $: productType, onChangeProductType(), clearError()
+  $: productId, onChangeProductId(), clearError()
+  $: issValue, clearError()
+  $: signOption, clearError()
```

Comment:
This is all very ugly, please use proper event handlers

Reply:
as in, `<input on:change={onChangeFoo()}>` and `clearError()` in those event handlers